src/services/CalculatorProcess.js

```diff
@@ -1,51 +1,50 @@
-let memory = [];
-let result = "";
-function calculatorProcess(content,visor) {
-	
-	const numbersOperators = () => String(eval((result)));
-	
-	memory += content;
+let memory = "";
+let sLength;
 
-	if (memory.includes("=")) {
+function CalculatorProcess(content, visor) {
 
-		result = memory.slice(0,-1);
+	const firstInteraction = ["%", "/", "*", "+", "-", "S", "C", "="];
 
-		if (memory.includes("+")) {
-			memory = numbersOperators();
-		}
+	memory += content;
 
-		if (memory.includes("-")) {
-			memory = numbersOperators();
-		}
+	if ((firstInteraction.includes(memory[0]) && memory.length < 2) || memory.includes("C")) {
+		memory = "";
 
-		if (memory.includes("*")) {
-			memory = numbersOperators();
-		}
+	}
+	
+	if (memory[0] == ".") {
+		memory = "0.";
+	}
 
-		if (memory.includes("/")) {
-			memory = numbersOperators();
-		}
+	if (memory.includes("=")) {
+
+		memory = memory.slice(0, -1);
 
 		if (memory.includes("%")) {
 			memory =
-    String(eval((memory[0]+memory[1]))*("0.0"+memory[3]));
+				String(eval(memory.replace(/[%]/g, "/100*")));
 		}
-	}
 
-	if (memory.includes("C")) {
-		memory = "";
+		memory = String(eval(memory));
```

Comment:
There is one thing redundante here

Reply:
It can be improved

Reply:
you mean because i'm receiving a string, converting it into a number and once again a string? or the fact i'm doing this conversion a lot ?

Reply:
i'm trying to remove this conversions and let just one or none